app/models/namespace.rb

```diff
@@ -120,6 +120,20 @@ def self.make_valid(name)
     name = final[0..MAX_NAME_LENGTH]
 
     return nil if name !~ NAME_REGEXP
+
+    # To avoid any name conflict we append an incremental number to the end
+    # of the name returns it as the name that will be used on both Namespace
+    # and Team on the User#create_personal_namespace! method
+    # TODO: workaround until we implement the namespace/team removal
+    if name =~ NAME_REGEXP
+      increment = 0
+      original_name = name
+      while Namespace.exists?(name: name)
+        name = "#{original_name}#{increment}"
```

Comment:
For the first time around, a `name0` will be created. I think it would be more elegant to have a `name` for the first case.

Reply:
Plus, just to avoid completely messed up scenarios, I'd give this loop a top value. That is:

```ruby
found = false
100.times do |idx|
  if Namespace.exists?(name: name)
    name = "#{original_name}#{idx + 1}"
  else
    found = true
    break
  end
end

# Return nil for crazy scenarios...
name if found
```

This would also preserve the name for the 0 case, as mentioned in my previous comment 😉

Reply:
> For the first time around, a name0 will be created. I think it would be more elegant to have a name for the first case.

It won't because it will not enter the `while` loop since `exists?` will return false. There's a test ensuring that what you said won't happen.

Reply:
Duh... makes sense 😁 That being said, could you put it inside of a `times` block ? just to be safe.

Reply:
Do you mean to avoid infinite loop if mysql/activerecord fails by never returning `true`?

Reply:
That, or even that the same user got removed and re-added 100 times 😁 (yeah, my pessimistic me is telling me that it will happen ...)